lib/src/components/AbstractDialog/AbstractDialog.tsx

```diff
@@ -0,0 +1,213 @@
+        // NOTE: this only works on listeners added after this one, which is believed to include any
+        // registered in useEffect calls on components rendered simultaneously with the dialog
+        else if (!hasNativeModalSupport) {
+          // emulate cancel-on-esc behavior in browsers which don't do it natively
+          onCancel(createCancelEvent());
+        }
+      }
+    }
+  }
+
+  useEffect(function() {
+    /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+    const el = dialogRef.current!,
+        // HTML <dialog> elements are supposed to remember what element was focused before they were opened,
+        // and restore focus to that element when they are closed. No browsers appear to implement this currently,
+        // so we do it ourselves
+        previouslyFocusedEl = document.activeElement;
+
+    setDialogRefState(el);
+
+    /*
+     * This will cause the document to become "blocked by the modal dialog"
+     * (https://html.spec.whatwg.org/multipage/interaction.html#blocked-by-a-modal-dialog)
+     * meaning that only the modal and its contents are interactable/focusable.
+     *
+     * Note: not supported in safari, which is why the conditional is here. Once safari gets
+     * it together, we should be able to take advantage of the "top layer" functionality built into
+     * the browser around modals to simplify the dialog styling around z-index handling
+     */
+    if (hasNativeModalSupport) {
+      // https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1029#issuecomment-968299542
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (el as any).showModal();
+    }
+    else {
+      // without native support we don't trap focus in the dialog, but we can at least start it off there
+      el.focus();
+    }
+
+    return () => {
+      if (previouslyFocusedEl && previouslyFocusedEl instanceof HTMLElement) {
+        // The useEffect cleanup executes while the dialog is still present (in React 16 at least). While the dialog
+        // still exists, the document is still "blocked by the modal dialog" so trying to focus elements outside of
+        // it won't work. So we have to wait until the next cycle of the event loop when it's gone
+        Promise.resolve().then(() => previouslyFocusedEl.focus());
+      }
+    };
+  }, []);
+
+  // listen to the native HTMLDialogElement cancel event which supporting browsers fire when the dialog is closed
+  // via ESC
+  useEffect(function() {
+    const dialog = dialogRef.current;
+
+    if (dialog && hasNativeModalSupport) {
+      /* eslint-disable @typescript-eslint/no-non-null-assertion */
+      dialog.addEventListener('cancel', onCancel!);
+      return () => { dialog.removeEventListener('cancel', onCancel!); };
+      /* eslint-enable @typescript-eslint/no-non-null-assertion */
+    }
+    else {
+      return undefined;
+    }
+  }, [onCancel]);
+
+  if (cancelOnClickOutside) {
```

Comment:
Since this behavior only pertains to `NxDrawer` and not `NxModal`, was it not possible to implement it within `NxDrawer`?

Reply:
> Since this behavior only pertains to `NxDrawer` and not `NxModal`, was it not possible to implement it within `NxDrawer`?

We moved it back here here:
fab5fe1ae780213af9632e1dad016203e89491b5
cc4b8c9c12a3c6d1a382598df68ac241e860ead8